pkg/segment/writer/segwriter.go

```diff
@@ -427,16 +438,32 @@ func timeBasedRotateSegment() {
 
 }
 
-func FlushWipBufferToFile(sleepDuration *time.Duration) {
+func FlushWipBufferToFile(idleWipFlushDuration *time.Duration, maxWaitWipFlushDuration *time.Duration) {
 	allSegStoresLock.RLock()
 	for streamid, segstore := range allSegStores {
 		segstore.Lock.Lock()
-		if segstore.wipBlock.maxIdx > 0 && time.Since(segstore.lastUpdated) > *sleepDuration {
+		if segstore.wipBlock.maxIdx == 0 {
+			segstore.Lock.Unlock()
+			continue
+		}
+
+		shouldFlush := false
+
+		if idleWipFlushDuration != nil && time.Since(segstore.lastUpdated) > *idleWipFlushDuration {
+			shouldFlush = true
+		}
+
+		if !shouldFlush && maxWaitWipFlushDuration != nil && time.Since(segstore.lastWipFlushTime) > *maxWaitWipFlushDuration {
```

Comment:
I think this function is handling both cases of the two pointers that are passed, should they do they are own checks and not worry about whether shouldFlush was set. basically :
`if idleWipFlushDuration != nil {
    check the time and set should flush accordingly
}
if  maxWaitWipFlushDuration != nil {
    check the time and set should flush accordingly
}
if shouldFlush {
 // do the flushing, it can come from any of those above two if checks, it does not matter because if 
// we are above the thresholds of each flush timer then we should flush it irrespective of the sleepDuration.
}
`

Reply:
ah I see makes sense. you can keep it then. Sorry I didn't catch that in the first read